src/components/MarkdownImage.tsx

```diff
@@ -15,12 +15,16 @@ const MarkdownImage = ({
   width,
   height,
   aspectRatio,
+  src,
   ...rest
 }: MarkdownImageProps) => {
   const imageAspectRatio = parseFloat(aspectRatio)
   let imageWidth = parseFloat(width)
   let imageHeight = parseFloat(height)
 
+  // Replace slashes from windows paths with forward slashes
+  const srcToString = src.toString()
```

Comment:
Because this `src` prop is typed from the Next `ImageProps` signature, it can accept either a string or an image data object, so `toString` is defining it as a string for TypeScript.

Might be better to transform this prop so there isn't the issue with Type errors or runtime if an image data object is passed to this component instead.

Reply:
Can add this little getter:
```ts
const getSrcString = () => {
  if (typeof src === "string") return src

  if ("default" in src) return src.default.src

  return src.src
}
```

Then
```tsx
<Link href={ getSrcString() } />
```

Reply:
I think that since it's for markdown pages only, `src` needs to be defined as a `string` only, and needs to be declared as such in the props type signature.

Reply:
Ah got it, no, it is fine. Makes sense now. I forgot that we were using in two places (href & src).

Reply:
All good!

SideNote: I could actually work seeing if TypeScript can be improved for the components being sent to the MDX provider, to ensure that props coming into the components match what is to be expected from the provider. (Basically `JSX.IntrinsicElements` I believe. Then there should no question as to what is getting sent to child components.

What do you think?